File: chapter03/myrobot1/src/forward.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <geometry_msgs/Twist.h>

int main(int argc, char** argv)
{
	ros::init(argc,argv,"forward");

	ros::NodeHandle n;
	ros::Publisher pub=n.advertise<geometry_msgs::Twist>("cmd_vel",10);

	int rate=10;
	float linear_velocity=0.2;
	float goal_distance=1.0;
	float time_duration;
	int ticks;
	
	time_duration=goal_distance/linear_velocity;
	ticks=rate*time_duration;

	geometry_msgs::Twist move_cmd;
	move_cmd.linear.x=linear_velocity;

	ros::Rate r(rate);

	for(int i=0;i<ticks;i++)
	{
		pub.publish(move_cmd);
		r.sleep();
	}

	move_cmd.linear.x=0.0;
	pub.publish(move_cmd);
	ros::Duration(1).sleep();
}

File: chapter03/myrobot1/scripts/forward.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String
from geometry_msgs.msg import Twist

class Forward():
	def __init__(self):
		rospy.init_node('forward',anonymous=True)
		self.pub=rospy.Publisher('cmd_vel',Twist,queue_size=10)
		rate=10
		r=rospy.Rate(rate)

		linear_velocity=0.2
		goal_distance=1
		time_duration=goal_distance/linear_velocity

		move_cmd=Twist()
		move_cmd.linear.x=linear_velocity
		ticks=int(rate*time_duration)
		for t in range(ticks):
			self.pub.publish(move_cmd)
			r.sleep()

		move_cmd=Twist()
		self.pub.publish(move_cmd)
		rospy.sleep(1)

if __name__=='__main__':
	try:
		Forward()
	except rospy.ROSInterruptException:
		rospy.loginfo("shutdown")